### PR TITLE
Update example branch name

### PR DESCRIPTION
**Pull request by MGibson1**
## Objective

Refining updates from #142.
- Missed example branch: updated to include a descriptive section

### CODE REVIEW

File: docs/contributing/pull-requests/branching.md [hunk 1]
```diff
@@ -14,14 +14,31 @@ assist in identifying and tracking down stale branches.
...
 
-In this example, the full branch name would be `Vault/PM-1234/update-csp-hashes`.
+In this example, the full branch name would be `vault/pm-1234/update-csp-hashes`.
+
+- Use only lower case letters in branch names.
+- Separate words with `-` and branch name sections with `/`. Only use these characters for word or
+  section separation.
+- Limit work description section to ~50 characters. Overall branch name should be ~80 characters,
+  max.
+- List of Teams. Teams with abbreviations _must_ use the abbreviation instead of the team name.
```
Comment (withinfocus) [inline docs/contributing/pull-requests/branching.md:31]: ⛏️ The "List of Teams." seems out of place and like a heading.

File: docs/contributing/pull-requests/branching.md [hunk 2]
```diff
@@ -14,14 +14,31 @@ assist in identifying and tracking down stale branches.
...
+  Teams without abbreviations _may not_ abbreviate:
+  - auth
+  - vault
+  - admin console (abbreviated to `ac`)
+  - secrets manager (abbreviated to `sm`)
+  - tools
+  - platform
+  - billing
+  - client integrations (abbreviated to `ci`)
+  - passwordless
```
Comment (withinfocus) [inline docs/contributing/pull-requests/branching.md:41]: ❓ Seems tenuous to maintain this. Is this level of detail needed?

File: docs/contributing/pull-requests/branching.md [hunk 3]
```diff
@@ -14,14 +14,31 @@ assist in identifying and tracking down stale branches.
 
 ### Branches for a specific Jira issue
 
-In order to tie the work on a branch to our Jira issues, the branch name should be made up of two
+In order to tie the work on a branch to our Jira issues, the branch name should be made up of three
```
Comment (differsthecat) [inline docs/contributing/pull-requests/branching.md:17]: ⛏️  'Link' flows better in this sentence to me
```suggestion
In order to link the work on a branch to our Jira issues, the branch name should be made up of three
```

File: docs/contributing/pull-requests/branching.md (hunk 2 above)
Comment (MGibson1) [reply docs/contributing/pull-requests/branching.md:41]: I'm fine removing, but I think _something_ should be used to encourage similar team name sections. if not for abbreviations I'd be less concerned

**Review by withinfocus — APPROVED** [review]
Thanks -- I like it.

**Pull request merged**